### PR TITLE
[GH-24] Add understandable documentation for non-developers

### PR DESCRIPTION
**Pull request by HirotoShioi**
Create an document explaining what the shell does for non-developers.

### CODE REVIEW

File: docs/Document.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+Various **features** are required for the cardano node to operate.
+
+* Blockchain
+* Ledger
+* Logging
+* Monitoring
+* Network
+* Wallet backend
+
+The shell will act as an **glue** of these features. It provides required resources/configurations to each of these features as well as resolving its dependencies.
```
Comment (denisshevchenko) [inline docs/Document.md]: > as an **glue**

as a **glue**, no?

Comment (HirotoShioi) [reply docs/Document.md]: Yes you're right. Will fix this.

File: docs/Document.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
+ # `cardano-shell` overview
+
+`cardano-shell` is an thin layer which brings all the other modules working
+ together and makes sure that they have the required resources 
+ (configuration, exception handling, monitoring, logging, ...).
+
+The shell is also responsible for:
```
Comment (ksaric) [inline docs/Document.md]: The shell is not responsible for logging and monitoring. That is brought in using the library, which is a part of shell (the part that can add features).

File: docs/Document.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+Various **features** are required for the cardano node to operate.
+
+* Blockchain layer
+* Ledger layer
+* Logging layer
+* Monitoring layer
+* Network layer
+* Wallet backend layer
+
+The shell will act as a **glue** of these features. It provides required
```
Comment (ksaric) [inline docs/Document.md]: The idea is that it will be a glue _for some of them_.

**Pull request merged**